tests/probes_test.go

```diff
@@ -103,11 +103,7 @@ var _ = Describe("[ref_id:1182]Probes", func() {
 		)
 
 		table.DescribeTable("should succeed when probing an IPv6 IP on dual stack clusters", func(readinessProbe *v12.Probe) {
-			isClusterDualStack, err := libnet.IsClusterDualStack(virtClient)
-			Expect(err).NotTo(HaveOccurred(), "should have been able to infer if the cluster is dual stack")
-			if !isClusterDualStack {
-				Skip("This test requires a dual stack network config.")
-			}
+			libnet.SkipIpv6TestsOnSingleStackCluster(virtClient)
```

Comment:
Forgive me for asking, but why are we skipping ?

I think this should work on single stack IPv6 - since the probe would default to the podIP, which, would be the IPv6 address.

Reply:
We don't support single stack ipv6 cluster anymore.
We test ipv6 only on dual stack cluster.

I thought about  changing `libnet.SkipIpv6TestsOnSingleStackCluster` to `ibnet.SkipIfClusterNotSupportIpv6` - which will skip the test if the cluster is not dual stack and the primary address is not ipv6. But I believe it may be misleading, since we don't support ipv6 single stack cluster at all.

We should assume that if the cluster always have ipv4 address (primary or not). We should fail otherwise.

Reply:
I am not sure we would want it, what about dev-script which can be an `External` provider with just ipv6 or just ipv4 ?
i feel its too early, and possibly wrong to stop supporting it.

Reply:
We don't have lane testing ipv6 only cluster == we don't support ipv6 only cluster.
Regarding ipv4 only, we do support ipv4 only cluster. Just ipv6 test should be skiped in case the cluster is single stack.

Reply:
> We don't support single stack ipv6 cluster anymore.
> We test ipv6 only on dual stack cluster.
> 
> I thought about changing `libnet.SkipIpv6TestsOnSingleStackCluster` to `ibnet.SkipIfClusterNotSupportIpv6` - which will skip the test if the cluster is not dual stack and the primary address is not ipv6. But I believe it may be misleading, since we don't support ipv6 single stack cluster at all.

I understand your point. I think this issue is 'bigger' than this PR, and we should first decide what we want to achieve.

Personally, I think that testing IPv6 only is so cheap that we should do it - but that raises the question of how to make explicit that this is not 'supported'. 

Your PR shouldn't be kept on hold until we decide what we want to do.

> 
> We should assume that if the cluster always have ipv4 address (primary or not). We should fail otherwise.